pkg/controllers/syncer/translator/namespaced_translator.go

```diff
@@ -214,16 +216,22 @@ func translateAnnotations(vObj client.Object, pObj client.Object, excluded []str
 	return retMap
 }
 
-func translateLabels(vObj client.Object) map[string]string {
+func translateLabels(vObj client.Object, syncedLabels []string) map[string]string {
 	newLabels := map[string]string{}
-	for k, v := range vObj.GetLabels() {
+	vObjLabels := vObj.GetLabels()
+	for k, v := range vObjLabels {
 		if k == translate.NamespaceLabel {
 			newLabels[k] = v
 			continue
 		}
 
 		newLabels[ConvertLabelKey(k)] = v
 	}
+	for _, k := range syncedLabels {
+		if value, ok := vObjLabels[k]; ok {
```

Comment:
This panics if the virtual object has no labels as the map is nil right?

Reply:
Good call again! I was assuming `GetLabels()` would not return nil but I can't verify that, so added this nil check.